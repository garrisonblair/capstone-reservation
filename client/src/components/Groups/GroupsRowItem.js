import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Table, Label } from 'semantic-ui-react';
import api from '../../utils/api';
import GroupsModal from './GroupsModal';
// import sweetAlert from 'sweetalert2';


class GroupsRowItem extends Component {
  state = {
    openModal: false,
    me: '',
  }

  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    api.getMyUser()
      .then((r) => {
        this.setState({ me: r.data.id });
      });
  }

  openModal = () => {
    this.setState({ openModal: true });
  }

  closeModal = () => {
    const { syncGroupsList } = this.props;
    syncGroupsList();
    this.setState({ openModal: false });
  }


  render() {
    const { group, syncGroupsList } = this.props;
    const { openModal, me } = this.state;
    return (
      <Table.Row key={group.id}>
        <Table.Cell>
          <h4>{group.name}</h4>
        </Table.Cell>
        <Table.Cell>
          {group.owner === me ? <Label color="yellow">ADMIN</Label> : <Label color="grey">MEMBER</Label>}
        </Table.Cell>
        <Table.Cell textAlign="center">
          <Button icon="edit" onClick={this.openModal} />
        </Table.Cell>
        {openModal ? (
          <GroupsModal
            show
            selectedGroup={group}
            onClose={this.closeModal}
            syncGroupsList={syncGroupsList}
          />
        ) : ''}
      </Table.Row>
    );
  }
}

GroupsRowItem.propTypes = {
  syncGroupsList: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  group: PropTypes.object.isRequired,
};

export default GroupsRowItem;
