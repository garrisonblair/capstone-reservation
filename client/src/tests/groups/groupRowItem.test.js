import React from 'react';
import renderer from 'react-test-renderer';
import GroupsRowItem from '../../components/Groups/GroupsRowItem';

function setup(myUserId, groupOwnerId) {
  const syncGroupsList = () => { };
  const group = {
    id: 1,
    name: 'groupName',
    owner: {
      id: groupOwnerId,
    },
  };

  return { syncGroupsList, group, myUserId };
}

it('Shows row as owner', () => {
  const { syncGroupsList, group, myUserId } = setup(1, 1);
  const tree = renderer
    .create(
      <GroupsRowItem
        syncGroupsList={syncGroupsList}
        group={group}
        myUserId={myUserId}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Shows row as member', () => {
  const { syncGroupsList, group, myUserId } = setup(1, 2);
  const tree = renderer
    .create(
      <GroupsRowItem
        syncGroupsList={syncGroupsList}
        group={group}
        myUserId={myUserId}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
