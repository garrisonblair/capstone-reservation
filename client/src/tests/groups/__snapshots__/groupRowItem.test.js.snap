// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows row as member 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    <h4>
      groupName
    </h4>
  </td>
  <td
    className=""
  >
    <div
      className="ui grey label"
      onClick={[Function]}
    >
      MEMBER
    </div>
  </td>
  <td
    className="center aligned"
  >
    <button
      className="ui icon button"
      onClick={[Function]}
      role="button"
    >
      <i
        aria-hidden="true"
        className="edit icon"
      />
    </button>
    
  </td>
</tr>
`;

exports[`Shows row as owner 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    <h4>
      groupName
    </h4>
  </td>
  <td
    className=""
  >
    <div
      className="ui yellow label"
      onClick={[Function]}
    >
      OWNER
    </div>
  </td>
  <td
    className="center aligned"
  >
    <button
      className="ui icon button"
      onClick={[Function]}
      role="button"
    >
      <i
        aria-hidden="true"
        className="edit icon"
      />
    </button>
    
  </td>
</tr>
`;
